### DISCUSSION TITLE
How can I use getServerState

### DISCUSSION THREAD

**Discussion by silverWolf818** [pmndrs/zustand#1994] [Q&A]
I want to pass getServerState in nextjs, how can I use getServerState

I found the code, but I don't know how to deliver it
```
api.getServerState || api.getState,
```

**Comment by dai-shi** [comment]
The real use case has been unknown, and your feedback would be appreciated.

Like the found code, you can specify `getServerState` like this:
```js
import { create } from 'zustand'

const useCountStore = create((set) => ({
  count: 0,
}))
useCountStore.getServerState = () => ({
  count: 1,
})
```
(But, there would be many tricky things, maybe.)

---

Wait, that doesn't work...
https://github.com/pmndrs/zustand/blob/5493959646e2d9dbca1feaad33e4593090e39227/src/react.ts#L111
We need to pass `useBoundStore` instead of `api` there.

Meanwhile, please try the following:
```js
import { createStore, useStore } from 'zustand'

const countStore = createStore((set) => ({
  count: 0,
}))
countStore.getServerState = () => ({
  count: 1,
})

const useCountStore = (selector) => useStore(countStore, selector)
```
